src/runtime/server/plugin.ts

```diff
@@ -109,19 +109,21 @@ export default defineNitroPlugin(async nitro => {
 
     let resolvedPath = undefined
     let redirectCode = 302
-    if (rootRedirect && event.path === '/') {
+    const requestURL = new URL(getRequestURL(event))
```

Comment:
The value returned by `getRequestURL` is already a `URL` so we should be able to use that directly.

```suggestion
    const requestURL = getRequestURL(event)
```